File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.java
class Solution {
    int mapValue(int[] mapping, int org) {
        int ans = 0, power = 1;
        if(org == 0) return mapping[0];
        while(org > 0) {
            ans = power * mapping[org%10] + ans;
            org = org/10;
            power = power * 10;
        }
        return ans;
    }
    public int[] sortJumbled(int[] mapping, int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        List<int[]> lp = new ArrayList<>();
        for(int org:nums) {
            int val = mapValue(mapping, org);
            lp.add(new int[] {val, org});
        }
        System.out.println(lp);
        Collections.sort(lp, (a, b) -> {
           if(a[0] == b[0]) return 0;
            return a[0] - b[0];
        });
        for(int i=0; i<n; i++) {
            ans[i] = lp.get(i)[1];
        }
        return ans;
    }
}

File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.cpp
class Solution {
public:
    int mapValue(int x, vector<int>& mapping) {
        if(x == 0) return mapping[0];
        string ans = "";
        while(x > 0) {
            ans += '0' + mapping[x%10];
            x = x/10;
        }
        reverse(ans.begin(), ans.end());
        return stoi(ans);
    }
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        vector<int>ans;
        vector<pair<int, int>>vp;
        for(int org:nums) {
            int x = mapValue(org, mapping);
            vp.push_back({x, org});
        }
        sort(vp.begin(), vp.end(), [&](pair<int, int>&A, pair<int, int>&B) {
            if(A.first == B.first) return false;
            else return A.first < B.first;
        });
        for(auto it:vp) {
            ans.push_back(it.second);
        }
        return ans;
    }
    static bool cmp(pair<int, int>&A, pair<int, int>&B) {
        return A.first == B.first ? false : A.first < B.first;
    }
};